src/HttpParser.jl

```diff
@@ -87,6 +87,59 @@ type ParserSettings
     on_chunk_complete::Ptr{Void}
 end
 
+function ParserSettings(on_message_begin_cb::Ptr{Void},
+  on_url_cb::Ptr{Void},
+  on_status_complete_cb::Ptr{Void},
+  on_header_field_cb::Ptr{Void},
+  on_header_value_cb::Ptr{Void},
+  on_headers_complete_cb::Ptr{Void},
+  on_body_cb::Ptr{Void},
+  on_message_complete_cb::Ptr{Void})
+    Base.depwarn("ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb)"
+            * " is deprecated use ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb,"
+            * " on_chunk_header, on_chunk_complete)", :ParserSettings)
+    function dummy(parser)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    ParserSettings(on_message_begin_cb, on_url_cb, on_status_complete_cb,
+      on_header_field_cb, on_header_value_cb, on_headers_complete_cb,
+      on_body_cb, on_message_complete_cb, dummy_cb, dummy_cb)
+end
+
+function ParserSettings(;
+  on_message_begin_cb::Ptr{Void} = C_NULL,
+  on_url_cb::Ptr{Void} = C_NULL,
+  on_status_complete_cb::Ptr{Void} = C_NULL,
+  on_header_field_cb::Ptr{Void} = C_NULL,
+  on_header_value_cb::Ptr{Void} = C_NULL,
+  on_headers_complete_cb::Ptr{Void} = C_NULL,
+  on_body_cb::Ptr{Void} = C_NULL,
+  on_message_complete_cb::Ptr{Void} = C_NULL,
+  on_chunk_header::Ptr{Void} = C_NULL,
+  on_chunk_complete::Ptr{Void} = C_NULL)
+    function dummy(x...)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    dummy_data_cb = cfunction(dummy, HTTP_DATA_CB...)
```

Comment:
> The CI says absolutely nothing if the code isn't tested at all.

Ya I overlooked that. Just tested it with 0.4.5 and it doesn't work (even though varargs are supported), so I need to fix it.

> LLVM has nothing to do with this. This is done at julia level and there's no way julia can reuse the code for the two dummies in this function and the one above. If one looks a little bit deeper one sees that `cfunction` actually checks the passed function for side effect.

You are right, this happens on the julia level, but I don't agree that the code can (theoretically) not be reused. If you look at this short example you will see that the generated c function pointer is always the same, which implies the opposite.

```
cfuncs = Ptr{Void}[]
funcs = Function[]

function blub()
  function bla()
    0
  end

  push!(cfuncs, cfunction(bla, Int, ()))
  push!(funcs, bla)
end

blub()
blub()

# 2-element Array{Ptr{Void},1}:
# Ptr{Void} @0x00007f811b134000
# Ptr{Void} @0x00007f811b134000
println(cfuncs)
```

Anyway, this is a theoretical discussion and looking at the `funcs` array in the above example shows that additional memory is allocated whenever `blub` is called, so I will move `dummy` outside of the constructor for performance reasons.